wordpress-mobile/WordPress-iOS
# Remove AccountService from all the things

Disclaimer: this PR description is written by @crazytonyli, not Jeremy. 😸 

----

## Changes

The main change is in [AccountService](https://github.com/wordpress-mobile/WordPress-iOS/pull/16168/files#diff-93852dcd12ccf58883495b486b5f8e5733039063979dea8f9a2f0932d22d3ef0), removing a few account lookup APIs from it. The rest changes are making the migration, using the APIs in [WPAccount+Lookup](https://github.com/wordpress-mobile/WordPress-iOS/blob/93d5bf4cf8f1853cae63a859ba0bf7159fab0070/WordPress/Classes/Models/WPAccount+Lookup.swift) instead. Even though there are lots of files are changed, but the changes are quite similar, I would recommend utilize the "Viewed" checkbox at top right of each file in the PR diff page—it sure helped me reviewing this PR.

## Test Instructions

Making sure features only available for logged in user are working as expected, and vise-versa. i.e. user is able to edit draft in their sites, not in others' sites.

PR submission checklist:

  _N/A. No new API is added in this PR_
  _N/A_
  _Not Needed_
